### PR TITLE
LPS-160493 AB Sidebar Panel does not get updated on SPA navigation

### PR DESCRIPTION
**Pull request by beltranrengifo**
[Jira ticket](https://issues.liferay.com/browse/LPS-160493)

## Motivation

After some refactoring, the mechanism that builds and updates the content of the Segments Experiment panel does not as often as it should.
The most noticeable issue is on SPA navigation, because the `URL` that the `side_navigation.es.js` module uses for [fetching the data automatically](https://github.com/liferay/liferay-portal/blob/278ea6228e43709cb67afa278db3cd348b619f32/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/side_navigation.es.js#L405) gets static 🤷 

## Proposed solution

As we did in other panels, we could move from this _black-box_ approach to a more open/transparent one. It feels the less problematic way and easier to debug in case of refactoring, etc.

By now, as today, we're just trying to make it work, using the `side_navigation.es.js` as is, but placing some workarounds to keep the url updated and living along with React.

## Video Recording of how to test it

https://user-images.githubusercontent.com/5776822/185906632-23f7487a-66bb-4825-ba07-bdaec44c074e.mp4

### CODE REVIEW

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
hi @cgoncas I'm not able to reproduce the bug anymore, could you try? thxs!

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:reevaluate:1297401_4169

**Comment by cgoncas** [comment]
Hey @beltranrengifo !!

It is still failing for me :(.

Thanks!!!

**Comment by cgoncas** [comment]
> Hey @beltranrengifo !!
> 
> It is still failing for me :(.
> 
> Thanks!!!

Silly me, I was not deploying the right module, looks like the bug is fixed. 

Amazing job @beltranrengifo !!!

**Review by cgoncas — APPROVED** [review]
LGTM

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:tango

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by michaelhashimoto** [comment]
ci:reevaluate:1297401_4210

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:tango

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
Hey @Luizcarlosqueiroz !!

Could you take a look at the TANGO failing test? I think they are related with this PR, but could you describe a step by step case where they are failing.

Thanks!!

**Comment by Luizcarlosqueiroz** [comment]
Hey @cgoncas !
The failures are associated with the PR changes, because the elements in the html have change, so the path is no longer working. I'll add a commit here to fix that.

**Comment by Luizcarlosqueiroz** [comment]
ci:test:tango

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:test:stop

**Comment by cgoncas** [comment]
ci:stop

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by Luizcarlosqueiroz** [comment]
A/B Test are passing again looking at latest [ci:test:relevant](https://testray.liferay.com/home/-/testray/case_results/index?p_p_state=normal&orderByCol=status_sortable&orderByType=asc&testrayBuildId=1958456266&testrayComponentId=1482768943&testrayRunId=1958458506).
image

**Comment by beltranrengifo** [comment]
hey @cgoncas !
what is the status of this pr?
should I rebase with master and hit the CI again?
is there anything else to do here?

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
> what is the status of this pr?

Ci:forward has been failing :(.

Trying to run it again.

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
ci:reevaluate:1317401_4792

**Comment by beltranrengifo** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:forward

**Comment by beltranrengifo** [comment]
ci:reevaluate:1317401_4811

**Comment by cgoncas** [comment]
ci:forward

**Comment by claraizquierdo** [comment]
ci:forward

File: modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/SegmentsExperimentApp.es.js [hunk 1]
```diff
@@ -61,50 +61,62 @@ export default function ({context, portletNamespace, props}) {
...
+								runSegmentsExperimentURL,
+							},
+							namespace: context.namespace,
+						}),
+						editVariantLayoutURL: editSegmentsVariantLayoutURL,
+						imagesPath,
+						page,
+					}}
+				>
+					<div id={`${portletNamespace}-segments-experiment-root`}>
```
Comment (claraizquierdo) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/SegmentsExperimentApp.es.js:90]: @beltranrengifo I see two divs with the same id 🤔 Should we remove this one?

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Pull request closed without merging**